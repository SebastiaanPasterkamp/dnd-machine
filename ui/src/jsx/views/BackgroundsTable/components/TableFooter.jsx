import React from 'react';

import BackgroundLinks from '../../../components/BackgroundLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={2}>
                <BackgroundLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
