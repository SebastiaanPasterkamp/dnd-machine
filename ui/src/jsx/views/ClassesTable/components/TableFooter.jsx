import React from 'react';

import ClassLinks from '../../../components/ClassLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={2}>
                <ClassLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
