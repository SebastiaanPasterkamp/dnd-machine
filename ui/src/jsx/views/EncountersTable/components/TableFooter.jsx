import React from 'react';

import EncounterLinks from '../../../components/EncounterLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={4}>
                <EncounterLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
