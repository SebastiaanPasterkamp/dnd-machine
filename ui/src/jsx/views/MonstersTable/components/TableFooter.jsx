import React from 'react';

import MonsterLinks from '../../../components/MonsterLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={4}>
                <MonsterLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
