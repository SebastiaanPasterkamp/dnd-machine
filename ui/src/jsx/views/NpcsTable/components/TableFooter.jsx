import React from 'react';

import NpcLinks from '../../../components/NpcLinks.jsx';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={7}>
                <NpcLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
