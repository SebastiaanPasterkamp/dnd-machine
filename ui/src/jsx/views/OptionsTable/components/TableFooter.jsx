import React from 'react';

import OptionLinks from '../../../components/OptionLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={2}>
                <OptionLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
