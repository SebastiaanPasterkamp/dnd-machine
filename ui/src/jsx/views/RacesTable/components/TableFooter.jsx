import React from 'react';

import RaceLinks from '../../../components/RaceLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={2}>
                <RaceLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
