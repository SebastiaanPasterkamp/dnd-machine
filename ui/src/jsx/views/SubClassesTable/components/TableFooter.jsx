import React from 'react';

import SubClassLinks from '../../../components/SubClassLinks';

export const TableFooter = function() {
    return (
        <tr>
            <td colSpan={2}>
                <SubClassLinks
                    altStyle={true}
                />
            </td>
        </tr>
    );
};

export default TableFooter;
