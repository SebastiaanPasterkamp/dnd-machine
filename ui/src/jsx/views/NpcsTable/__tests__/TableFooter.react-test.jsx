import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MockRouter from 'react-mock-router';

import ListDataActions from 'actions/ListDataActions';
import { mockedApi } from '../../../../../tests/__mocks__';

import TableFooter from '../components/TableFooter';

describe('TableFooter', () => {

    beforeEach(() => {
        fetch.mockImplementation( mockedApi({
            current_user: {
                id: 1,
                role: ['dm'],
            },
        }) );
        ListDataActions.fetchItems('current_user');
    })

    afterEach(() => fetch.resetMocks())

    it('should render', () => {
        const tree = renderer.create(
            <MockRouter>
                <TableFooter />
            </MockRouter>
        );

        expect(tree).toMatchSnapshot();
    });

});
