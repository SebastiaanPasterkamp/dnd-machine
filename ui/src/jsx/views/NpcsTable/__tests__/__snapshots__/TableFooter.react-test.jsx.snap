// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFooter should render 1`] = `
<tr>
  <td
    colSpan={6}
  >
    <div
      className="base-link-group nice-btn-group"
    >
      <a
        className="cursor-pointer nice-btn-alt icon fa-plus"
        href={undefined}
        onClick={[Function]}
      >
        New
      </a>
    </div>
  </td>
</tr>
`;
